tests/cupy_tests/sorting_tests/test_search.py

```diff
@@ -278,6 +278,38 @@ def test_flatnonzero(self, xp, dtype):
         return xp.flatnonzero(array)
 
 
+@testing.parameterize(
+    {'array': numpy.random.randint(0, 2, (20,))},
+    {'array': numpy.random.randn(3, 2, 4)},
+    {'array': numpy.empty((0,))},
+    {'array': numpy.empty((0, 2))},
+    {'array': numpy.empty((0, 2, 0))},
+)
+@testing.gpu
+class TestArgwhere(unittest.TestCase):
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_argwhere(self, xp, dtype):
+        array = xp.array(self.array, dtype=dtype)
+        return xp.argwhere(array)
```

Comment:
Not sure what happened with this test script, but I don't think the test style here is compliant with other tests. Usually we parametrize over shapes, and then initialize the input on CPU (numpy) or GPU (cupy) with the parameters to avoid explicit data transfer. For random input and other kinds of inputs, we can have different test methods (say `test_argwhere_random`, `test_argwhere_empty`, or just `test_argwhere_1`, `..._2`, etc).

For this case, though, it is @asi1024's call. We might need a separate PR for improving the style?

Reply:
Yes, we prefer to parametrize over shapes, but I think we don't need to resolve it in this PR.